### PR TITLE
Test ostream

### PR DESCRIPTION
**Pull request by vladpirlog**
- added file for constructors

- overloaded the assignment operator

- made small changes to the implementation of << and >> operators

- added tests for ostream operator

### CODE REVIEW

File: src/bigint.cpp
Comment (ayaankhan98) [inline src/bigint.cpp:7]: ctors are also linked with #16 , but the regex validation for string parameter is missing there, i will merge this after merging #16 

after merging #16 you should modify this file in accordance with #16 inorder to avoid merge conflicts.

File: src/bigint.hpp
Comment (ayaankhan98) [inline src/bigint.hpp:34]: please keep following this
```C++
void function () {
}
namespace {
}
int main () {
}
```
pattern instead of
```C++
int main ()
{
}
void function() 
{
}
```
this pattern.
soon we will add contribution and coding guidelines to this repository

File: src/operators/assignment_operator.cpp
Comment (ayaankhan98) [inline src/operators/assignment_operator.cpp:35]: modify the declaration of this function to recieve a `const largeInt&` instead of `largeInt`, this will avoid copying the whole string, just a bit of optimization

Comment (ayaankhan98) [reply src/operators/assignment_operator.cpp:35]: and please change the formatting scheme, rest everythiing looks good to me for now in this file

File: src/operators/istream_operator.cpp
Comment (ayaankhan98) [inline src/operators/istream_operator.cpp:34]: please change the formatting schem

File: src/operators/ostream_operator.cpp
Comment (ayaankhan98) [inline src/operators/ostream_operator.cpp:34]: please change the formatting schem

**Review by ayaankhan98 — COMMENTED** [review]
@vladpirlog Thanks for your contribution, please have a look at few suggestions

**Comment by ayaankhan98** [comment]
project structure is reconfigured and automated test and build process is also enabled now, please resolve the merge conflicts

File: src/bigint.hpp
Comment (ayaankhan98) [inline src/bigint.hpp:45]: Codacy found an issue: [Class 'largeInt' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

**Comment by ayaankhan98** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291&bid=20451770)

File: src/bigint.hpp
Comment (ayaankhan98) [inline src/bigint.hpp:47]: Codacy found an issue: [Class 'largeInt' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

**Comment by ayaankhan98** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
- Added 5
           

Clones added
============
- big-int/src/bigint.cpp  1
- src/bigint.cpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291&bid=20451770)

File: big-int/src/bigint.hpp
Comment (ayaankhan98) [inline big-int/src/bigint.hpp:43]: Codacy found an issue: [Class 'largeInt' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

File: big-int/src/bigint.hpp
Comment (ayaankhan98) [inline big-int/src/bigint.hpp:44]: Codacy found an issue: [Class 'largeInt' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

File: big-int/src/bigint.hpp
Comment (ayaankhan98) [inline big-int/src/bigint.hpp:42]: Codacy found an issue: [Class 'largeInt' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

File: big-int/src/bigint.hpp
Comment (ayaankhan98) [inline big-int/src/bigint.hpp:45]: Codacy found an issue: [Class 'largeInt' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

File: big-int/src/bigint.hpp
Comment (ayaankhan98) [inline big-int/src/bigint.hpp:101]: Codacy found an issue: ['largeInt::operator=' should return 'largeInt &'.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

**Comment by ayaankhan98** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 4
           

Clones added
============
- src/bigint.cpp  1
- big-int/src/bigint.cpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291&bid=20451770)

File: big-int/src/bigint.hpp
Comment (ayaankhan98) [inline big-int/src/bigint.hpp:103]: Codacy found an issue: ['largeInt::operator=' should return 'largeInt &'.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

File: big-int/src/bigint.hpp
Comment (ayaankhan98) [inline big-int/src/bigint.hpp:45]: Codacy found an issue: [Class 'largeInt' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

File: big-int/src/bigint.hpp
Comment (ayaankhan98) [inline big-int/src/bigint.hpp:46]: Codacy found an issue: [Class 'largeInt' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

File: big-int/src/bigint.hpp
Comment (ayaankhan98) [inline big-int/src/bigint.hpp:44]: Codacy found an issue: [Class 'largeInt' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291)

**Comment by ayaankhan98** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           

Clones added
============
- big-int/src/bigint.cpp  1
- src/bigint.cpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291&bid=20451770)

**Comment by ayaankhan98** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           

Clones added
============
- src/bigint.cpp  1
- big-int/src/bigint.cpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291&bid=20451770)

**Comment by ayaankhan98** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291&bid=20451770)

**Comment by ayaankhan98** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6285291&bid=20451770)

**Comment by ayaankhan98** [comment]
@vladpirlog tests for ostream operator are not correctly working

File: big-int/src/bigint.cpp
Comment (ayaankhan98) [inline big-int/src/bigint.cpp:25]: this regex expression does not consider letters 'a-z' && 'A-Z' and other symbols. in order words only number 0-9 should be allowed, if input is other than 0-9 through an exception `invalid integer: integer must contains digits from 0-9`

File: big-int/src/operators/assignment_operator.cpp
Comment (ayaankhan98) [inline big-int/src/operators/assignment_operator.cpp:35]: please include test for assignment operator

File: big-int/src/bigint.cpp
Comment (ayaankhan98) [inline big-int/src/bigint.cpp:7]: include tests for ctors

File: big-int/src/bigint.cpp
Comment (vladpirlog) [reply big-int/src/bigint.cpp:25]: The regex matches any string which may or may not start with a "-", followed by at least one digit. Nothing else will go through. The function `std::regex_match` does a full string match, not a search for that pattern.

If you want, I could add tests for that regex pattern too.

**Review by ayaankhan98 — APPROVED** [review]
LGTM.
Thanks for your contribution

**Comment by ayaankhan98** [comment]
#34

**Pull request merged**
